packages/kbn-es-query/src/filters/contains.js

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Creates an filter where the given field contains the given value
+export function buildContainsFilter(field, value, indexPattern) {
+  const index = indexPattern.id;
+  const type = 'contains';
+  const key = field.name;
+
+  const filter = {
+    meta: { index, type, key, value },
+  };
+  filter.query = {
```

Comment:
One thing we learned when we were working on adding wildcard support to KQL is that `text` fields don't work well with simple wildcard queries. It's actually best to use a query_string query for wildcard queries on `text` fields because it does some special handling that makes it work how most users would expect. I'd suggest updating the query building logic to look like this:

```
  // Text fields should be treated in a special manner because their values are analyzed.
  // For example, a field configured with the standard analyzer with a value of "Foo Bar" would not
  // match with a simple wildcard query on the value "Foo Ba*" because the analyzer tokenized and
  // lowercased the text at index time. The actual values stored in the inverted index are "foo" and
  // "bar" but the wildcard query is literally searching for a token that starts with "Foo Ba".
  // The query_string query attempts to make wildcards more useful with text fields. It will do
  // tokenization and some simple normalization on the term that contains the wildcard, so that the
  // query "Foo Ba*" would actually match. This functionality is not exposed in any other query type
  // so we have to sort of abuse the query_string query here if we're using a text field
  if (field.esTypes && field.esTypes.includes('text')) {
    filter.query = {
      query_string: {
        fields: [field.name],
        query: `*${escapeQueryString(value)}*`,
      }
    };
  }
  else {
    filter.query = {
      wildcard: {
        [field.name]: {
          value: `*${value}*`,
        },
      },
    };
  }
```